plugins/shrink-path/shrink-path.plugin.zsh

```diff
@@ -93,7 +94,13 @@ shrink_path () {
                                 print '  zstyle :prompt:shrink_path fish yes'
                                 return 0
                         ;;
-                        -l|--last) lastfull=1 ;;
+                        -l|--last)
+                          lastfull=1
+                          if [[ "$2" =~ ^[0-99]+$ ]]; then
```

Comment:
regex comparison is slow in zsh, we can use a faster check for non-numerical characters:
```suggestion
                          if [[ -n "$2" && "$2" != *[^0-9]* ]]; then
```